src/test/clojure/jarl/builtins/encoding_test.clj

```diff
@@ -0,0 +1,80 @@
+(ns jarl.builtins.encoding_test
+  (:require [clojure.test :refer [deftest is testing]]
+            [jarl.builtins.encoding :refer [builtin-base64-encode builtin-base64-decode builtin-base64-url-encode
+                                            builtin-base64-url-encode-no-pad builtin-base64-url-decode
+                                            builtin-url-query-encode builtin-url-query-decode builtin-json-unmarshal
+                                            builtin-json-is-valid builtin-hex-encode builtin-hex-decode]]))
+
+(deftest builtin-base64-encode-test
+  (testing "base64.encode"
+    (is (= (builtin-base64-encode "abc123") "YWJjMTIz"))
+    (is (= (builtin-base64-encode "målarfärg") "bcOlbGFyZsOkcmc="))
+    (is (= (builtin-base64-encode "The Rego Playground") "VGhlIFJlZ28gUGxheWdyb3VuZA=="))
+    (is (= (builtin-base64-encode "😆") "8J+Yhg=="))))
+
+(deftest builtin-base64-decode-test
+  (testing "base64.decode"
+    (is (= (builtin-base64-decode "YWJjMTIz") "abc123"))
+    (is (= (builtin-base64-decode "bcOlbGFyZsOkcmc=") "målarfärg"))
+    (is (= (builtin-base64-decode "VGhlIFJlZ28gUGxheWdyb3VuZA==") "The Rego Playground"))
+    (is (= (builtin-base64-decode "8J+Yhg==") "😆"))))
+
+(deftest builtin-base64-url-encode-test
+  (testing "base64url.encode"
+    (is (= (builtin-base64-url-encode "abc123") "YWJjMTIz"))
+    (is (= (builtin-base64-url-encode "målarfärg") "bcOlbGFyZsOkcmc="))
+    (is (= (builtin-base64-url-encode "The Rego Playground") "VGhlIFJlZ28gUGxheWdyb3VuZA=="))
```

Comment:
OPA doesn't drop the `=` in the base64url encoding?

Reply:
Nope! Only when the `no_pad` variant is used:
```shell
opa eval -f pretty 'base64url.encode("a")'
"YQ=="
```